tests/bdd/features/snapshot/delete/delete.feature

```diff
@@ -0,0 +1,73 @@
+Feature: Volume Snapshot deletion
+
+  Background:
+    Given a deployer cluster
+
+  # USUAL DELETE OPERATION
+  Scenario Outline: Snapshot deletion volume
+    Given a <publish_status> volume with single replica
+    And a snapshot is created for the volume
+    When the created snapshot is deleted
+    Then the snapshot deletion should not fail
+    And the snapshot should not be present upon listing
+    Examples:
+      | publish_status |
+      | published      |
+      | unpublished    |
+
+
+  # USUAL DELETE OPERATION WITHOUT SOURCE
+  Scenario Outline: Snapshot deletion volume after volume deletion
+    Given a <publish_status> volume with single replica
+    And a snapshot is created for the volume
+    And the source volume is deleted
+    When the created snapshot is deleted
+    Then the snapshot deletion should not fail
+    And the snapshot should not be present upon listing
+    Examples:
+      | publish_status |
+      | published      |
+      | unpublished    |
+
+
+  # SOURCE DELETION WHILE IN PROGRESS (could be flaky ?)
+  Scenario Outline: Remove published volume source while snapshot creation is in progress
+    Given a <publish_status> volume with single replica
+    And a snapshot is created for the volume and the creation is in progress
+    When the source volume is deleted
+    Then the volume should not get deleted
```

Comment:
Interesting, I don't think we currently check if we have snapshots in creating, but I don't think this should block volume deletion, since creating may stay in creating forever...

Reply:
This is why I asked about the how..
I don't think we should be testing this with concurrent rpc calls in a bdd, it's not the place for it.
What we can test is try to create a snapshot, fails but stays in creating...
then delete the volume!! what should we do here, accept the delete and set the snapshot to deleting? I would say so..

Reply:
Right. Removing this for now.